src/aclk/mqtt_websockets/mqtt_ng.c

```diff
@@ -620,54 +591,30 @@ void transaction_buffer_transaction_rollback(struct transaction_buffer *buf, str
 struct mqtt_ng_client *mqtt_ng_init(struct mqtt_ng_init *settings)
 {
     struct mqtt_ng_client *client = callocz(1, sizeof(struct mqtt_ng_client));
-    if (client == NULL)
-        return NULL;
 
-    if (transaction_buffer_init(&client->main_buffer, HEADER_BUFFER_SIZE))
-        goto err_free_client;
+    transaction_buffer_init(&client->main_buffer, HEADER_BUFFER_SIZE);
 
     client->rx_aliases = RX_ALIASES_INITIALIZE();
-    if (client->rx_aliases == NULL)
-        goto err_free_trx_buf;
 
-    if (pthread_mutex_init(&client->stats_mutex, NULL))
-        goto err_free_rx_alias;
+    spinlock_init(&client->stats_spinlock);
+    spinlock_init(&client->tx_topic_aliases.spinlock);
 
     client->tx_topic_aliases.stoi_dict = TX_ALIASES_INITIALIZE();
-    if (client->tx_topic_aliases.stoi_dict == NULL)
-        goto err_free_stats_mutex;
     client->tx_topic_aliases.idx_max = UINT16_MAX;
 
-    if (pthread_rwlock_init(&client->tx_topic_aliases.rwlock, NULL))
-        goto err_free_tx_alias;
-
     // TODO just embed the struct into mqtt_ng_client
     client->parser.received_data = settings->data_in;
     client->send_fnc_ptr = settings->data_out_fnc;
     client->user_ctx = settings->user_ctx;
 
-    client->log = settings->log;
-
     client->puback_callback = settings->puback_callback;
     client->connack_callback = settings->connack_callback;
     client->msg_callback = settings->msg_callback;
 
     return client;
-
-err_free_tx_alias:
-    c_rhash_destroy(client->tx_topic_aliases.stoi_dict);
-err_free_stats_mutex:
-    pthread_mutex_destroy(&client->stats_mutex);
-err_free_rx_alias:
-    c_rhash_destroy(client->rx_aliases);
-err_free_trx_buf:
-    transaction_buffer_destroy(&client->main_buffer);
-err_free_client:
-    freez(client);
-    return NULL;
 }
 
-static inline uint8_t get_control_packet_type(uint8_t first_hdr_byte)
+static uint8_t get_control_packet_type(uint8_t first_hdr_byte)
```

Comment:
Considering the contention inside braces, probably we could keep the `inline` here.

Reply:
Not needed since its simple enough (and static) and the compiler will inline it